### PR TITLE
Add details for flash update

### PR DESCRIPTION
**Pull request by barshathakuri**
## Changes
- Add RichTextOutput Component
- Add Flash Update details
- Add Styling of Flash Update

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/FlashUpdateDetails/index.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                                countryRoute.absolutePath,
+                                {
+                                    countryId: country
+                                        ?.country_details.id,
+                                },
+                            ) : undefined}
+                    >
+                        {flashUpdateResponse?.country_district?.map(
+                            (district) => district.country_details?.name,
+                        )}
```
Comment (samshara) [inline src/views/FlashUpdateDetails/index.tsx]: we should join the strings with `,`

File: src/views/FlashUpdateDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                        >
+                            {flashUpdateResponse.map_files.map((item) => (
+                                <div
+                                    className={styles.mapItem}
+                                    key={item.id}
+                                >
+                                    <img
+                                        className={styles.image}
+                                        src={item.file}
+                                        alt=""
```
Comment (samshara) [inline src/views/FlashUpdateDetails/index.tsx]: can't we use approprite `alt`

**Review by samshara — APPROVED** [review]

File: src/views/FlashUpdateDetails/index.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+    const {
+        response: flashUpdateResponse,
+        pending: flashUpdatePending,
+    } = useRequest({
+        skip: isNotDefined(flashUpdateId),
+        url: '/api/v2/flash-update/{id}/',
+        pathVariables: {
+            id: Number(flashUpdateId),
+        },
```
Comment (tnagorra) [inline src/views/FlashUpdateDetails/index.tsx]: Let's check if flashUpdateId defined before we cast to number

File: src/views/FlashUpdateDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                    <Button
+                        name="Export"
+                    >
+                        {strings.flashUpdateExport}
+                    </Button>
+                    <Button
+                        name="Share"
+                    >
+                        {strings.flashUpdateShare}
+                    </Button>
```
Comment (tnagorra) [inline src/views/FlashUpdateDetails/index.tsx]:
```suggestion
                    <Button
                        name="export"
                    >
                        {strings.flashUpdateExport}
                    </Button>
                    <Button
                        name="share"
                    >
                        {strings.flashUpdateShare}
                    </Button>
```

File: src/views/FlashUpdateDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                    <Link
+                        to={country?.country_details
+                            ? generatePath(
+                                countryRoute.absolutePath,
+                                {
+                                    countryId: country
+                                        ?.country_details.id,
+                                },
+                            ) : undefined}
+                    >
```
Comment (tnagorra) [inline src/views/FlashUpdateDetails/index.tsx]: Let's add key prop

File: src/views/FlashUpdateDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+                        >
+                            {flashUpdateResponse?.graphics_files?.map((item) => (
+                                <div
+                                    key={item.id}
+                                    className={styles.graphicItem}
+                                >
+                                    <img
+                                        className={styles.image}
+                                        src={item.file}
+                                        alt=""
```
Comment (tnagorra) [inline src/views/FlashUpdateDetails/index.tsx]: Let's pass appropriate alt text

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/FlashUpdateDetails/index.tsx [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+        flashUpdateFormEdit: flashUpdateFormEditRoute,
+    } = useContext(RouteContext);
+
+    const hasActions = flashUpdateResponse?.actions_taken?.some(
+        (at) => (at?.actions?.length !== 0 || at.summary),
+    );
+
+    const districtName = flashUpdateResponse?.country_district?.map(
+        (district) => district.country_details?.name,
+    );
```
Comment (tnagorra) [inline src/views/FlashUpdateDetails/index.tsx]: Let's memoize these

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/FlashUpdateDetails/index.tsx [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+    });
+
+    const {
+        country: countryRoute,
+        flashUpdateFormEdit: flashUpdateFormEditRoute,
+    } = useContext(RouteContext);
+
+    const hasActions = useMemo(
+        () => flashUpdateResponse?.actions_taken?.map(
+            (at) => at?.actions?.length !== 0 || at.summary,
```
Comment (AdityaKhatri) [inline src/views/FlashUpdateDetails/index.tsx]:
```suggestion
            (action) => action?.actions?.length !== 0 || action.summary,
```

File: src/views/FlashUpdateDetails/index.tsx [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+                                        <Button
+                                            variant="secondary"
+                                            className={styles.downloadLink}
+                                            name={undefined}
+                                        >
+                                            {strings.flashUpdateDownloadDocument}
+                                        </Button>
+                                    ) : (
+                                        <div className={styles.notDownloadLink} />
+                                    )}
```
Comment (AdityaKhatri) [inline src/views/FlashUpdateDetails/index.tsx:245]: Fix UI inconsistencies

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Fix links in flash update.
Add navigational links to flash update view and edit pages in flash update list page.

Flash update listing page's url is `flash-updates/all` and invidual is `flash-update/{id}` we need to make this consistent as well.

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**